presto-main/src/test/java/com/facebook/presto/operator/scalar/TestHyperLogLogFunctions.java

```diff
@@ -0,0 +1,183 @@
+
+        double error = uniqueElts * 0.05;
+        assertNull(getCardinality(projection));
+    }
+
+    @Test
+    public void testApproxSingleColumn()
+    {
+        long blockSize = 1;
+        long uniqueElts = 10000 * blockSize;
+
+        String projection = getProjectionString(getUniqueElements((int) blockSize, uniqueElts));
+
+        double error = uniqueElts * 0.05;
+        assertEquals(getCardinality(projection), uniqueElts, error);
+    }
+
+    @Test
+    public void testApproxTwoColumns()
+    {
+        long blockSize = 2;
+        long uniqueElts = 10000 * blockSize;
+
+        String projection = getProjectionString(getUniqueElements((int) blockSize, uniqueElts));
+
+        double error = uniqueElts * 0.05;
+        assertEquals(getCardinality(projection), uniqueElts, error);
+    }
+
+    @Test
+    public void testApproxThreeColumns()
+    {
+        long blockSize = 3;
+        long uniqueElts = 10000 * blockSize;
+
+        String projection = getProjectionString(getUniqueElements((int) blockSize, uniqueElts));
+
+        double error = uniqueElts * 0.05;
+        assertEquals(getCardinality(projection), uniqueElts, error);
+    }
+
+    @Test
+    public void testApproxManyColumns()
+    {
+        //max number of columns to merge is 254
+        long blockSize = 254;
+        long uniqueElts = 1000 * blockSize;
+        String projection = getProjectionString(getUniqueElements((int) blockSize, uniqueElts));
+
+        double error = uniqueElts * 0.05;
+        assertEquals(getCardinality(projection), uniqueElts, error);
+    }
+
+    private Long getCardinality(String projection)
+    {
+        //variables from FunctionAssertions.java to get results of query
+        Session session = SessionTestUtils.TEST_SESSION;
+        FeaturesConfig featuresConfig = new FeaturesConfig();
+        LocalQueryRunner runner = new LocalQueryRunner(session, featuresConfig);
+
+        MaterializedResult result = runner.execute("SELECT " + projection);
+        //will return a Long, since that's the return type for cardinality()
+        Long cardinality = (Long) result.getOnlyValue();
+
+        return cardinality;
+    }
+
+    private HyperLogLog[] getUniqueElements(int blockSize, long uniqueElements)
+    {
+        HyperLogLog[] array = new HyperLogLog[blockSize];
+
+        for (int j = 0; j < blockSize; j++) {
+            //creates a HyperLogLog a single column
+            HyperLogLog firstHll = HyperLogLog.newInstance(NUMBER_OF_BUCKETS);
+
+            //populates column with even partitions of the unique elements
+            for (long i = j * uniqueElements / blockSize; i < j * uniqueElements / blockSize +
+                    uniqueElements / blockSize; i++) {
+                firstHll.add(i);
+            }
+
+            array[j] = firstHll;
+        }
+
+        return array;
+    }
+
+    private String getProjectionString(HyperLogLog... array)
```

Comment:
Same as above. I would use `List` instead of `array`. But I don't have a strong opinion here.